File: src/providers/notifications/slack.py
"""Slack notification adapter."""

import logging
from src.providers.base import NotificationBase

logger = logging.getLogger(__name__)


class SlackAdapter(NotificationBase):
    name = "slack"
    label = "Slack"

    async def send(self, message, config):
        channel = config.get("channel", "general")
        logger.info(f"[Slack -> #{channel}] {message}")


adapter = SlackAdapter()

File: src/providers/notifications/slack.js
/**
 * @module providers/notifications/slack
 * @description Slack notification adapter.
 *
 * Sends notifications to a Slack channel via the Slack MCP or
 * by logging the message (when MCP is not available).
 */

const { NotificationBase } = require('../base/notification-base');
const logger = require('../../utils/logger');

class SlackAdapter extends NotificationBase {
  get name() { return 'slack'; }
  get label() { return 'Slack'; }

  async send(message, config) {
    const channel = config.channel || 'general';
    // In the agent context, Slack MCP handles delivery.
    // From the webhook server context, we log for now.
    // A full implementation would call the Slack API directly.
    logger.info(`[Slack -> #${channel}] ${message}`);
  }
}

module.exports = new SlackAdapter();
